### ISSUE TITLE
[Question] does custom CNN feature extractor work? 

### ISSUE THREAD

**Issue by gglsmm** [DLR-RM/stable-baselines3#880]
**Important Note: We do not do technical support, nor consulting** and don't answer personal questions per email.
Please post your question on the [RL Discord](https://discord.com/invite/xhfNqQv), [Reddit](https://www.reddit.com/r/reinforcementlearning/) or [Stack Overflow](https://stackoverflow.com/) in that case.

### Question

I use stable-baselines3 and train an agent with PPO for the Pong game. I use a custom CNN feature extractor instead of NatureCNN. But it seems the model does not learn. I followed the documentation of stable-baselines3 to create a custom CNN.

Currently, the model is at the 9m timesteps and the episode reward mean is about -20.5. is there something wrong with the feature extractor part? Thank you in advance
### Additional context
```python
class CustomCNN(BaseFeaturesExtractor):
"""
:param observation_space: (gym.Space)
:param features_dim: (int) Number of features extracted.
    This corresponds to the number of unit for the last layer.
"""

def __init__(self, observation_space: gym.spaces.Box, features_dim: int = 512):
    super(CustomCNN, self).__init__(observation_space, features_dim)
    # We assume CxHxW images (channels first)
    # Re-ordering will be done by pre-preprocessing or wrapper
    n_input_channels = observation_space.shape[0]
    self.cnn = nn.Sequential(
        nn.Conv2d(n_input_channels, 128, kernel_size=2, stride=1, padding=0),
        nn.GELU(),
        nn.Conv2d(128, 16, kernel_size=2, stride=1, padding=0),
        nn.ReLU(),
        nn.Conv2d(16, 32, kernel_size=8, stride=1, padding=0),
        nn.CELU(),
        nn.Flatten(),
    )
    with th.no_grad():
        n_flatten = self.cnn(
            th.as_tensor(observation_space.sample()[None]).float()
        ).shape[1]

    self.linear = nn.Sequential(nn.Linear(n_flatten, features_dim), nn.ReLU())

def forward(self, observations: th.Tensor) -> th.Tensor:
    return self.linear(self.cnn(observations))


policy_kwargs = dict(
features_extractor_class=CustomCNN,
features_extractor_kwargs=dict(features_dim=512),
)

model = PPO("CnnPolicy", env, policy_kwargs=policy_kwargs,)
model.learn(10000000)
```

### Checklist

- [ ] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [ ] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

**Comment by Miffyli** [comment]
Hey. We do not unfortunately have time to offer custom tech support like this ("why is it not learning?"). We can only be certain for the results in the docs and [zoo](https://github.com/DLR-RM/rl-baselines3-zoo/). Any modifications on them is in unknown territories for us.

My recommendation is that you take the code in the zoo library for running Pong, run it to ensure you get expected results (learning), and then start modifying it bit-by-bit. There are _many_ small things that could be ruining the results in your code, so it is better to take an established starting point and go from there.

Closing as "no tech support". See links at the top of your post for potential places for people who could potentially help you out.
